hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -473,6 +475,19 @@ private void handleEndInputEvent(WriteMetadataEvent event) {
     }
   }
 
+  private void filterWriteFailure(WriteMetadataEvent event) throws HoodieException {
+    // It will early detect any write failures in each of task and rollback the instant
+    // to prevent data loss caused by commit failure after a checkpoint is finished successfully.
+    if (!event.isBootstrap()) {
```

Comment:
why bootstrap is excluded, early failure can also be triggered in this case?

Reply:
Bootstrap event doesn't have write statuses.

Reply:
iiuc, bootstrap event restored from state may have write status

Reply:
I see. That is I were not aware of. Fixed. Thanks.